File: katt/magic/submissions/accepted/magic.java
import java.util.*;

class magic {
	public static int magic_score(int N, int K, int[] L, int[] R) {
		int[] ds = new int[N];
		for (int i = 0; i < N; i++) {
			assert R[i] - L[i] == 2;
			ds[i] = Math.min(Math.abs(L[i]), Math.abs(R[i]));
			grader.lib.trick(0);
		}
		Arrays.sort(ds);
		int res = 0;
		for (int x : ds) {
			if (x <= K) {
				K -= x;
				res++;
			}
		}
		return res;
	}
}

File: katt/magic/submissions/accepted/magic.py
import grader

def magic_score(N, K, L, R):
    ds = []
    for i in range(N):
        assert R[i] - L[i] == 2
        ds.append(min(abs(L[i]), abs(R[i])))
        grader.trick(0)
    res = 0
    ds.sort()
    for x in ds:
        if x <= K:
            K -= x
            res += 1
    return res
